backend/src/Edna/DB/Schema.hs

```diff
@@ -1,39 +1,331 @@
+    deriving stock (Generic)
+    deriving anyclass Beamable
+  primaryKey = TestMethodologyId . tmTestMethodologyId
+
+deriving stock instance Show (PrimaryKey TestMethodologyT Identity)
+deriving stock instance Eq (PrimaryKey TestMethodologyT Identity)
+
+--------------------------
+-- Target
+--------------------------
+
+data TargetT f = TargetRec
+  { tTargetId :: C f (SqlSerial Int32)
+  , tName :: C f Text
+  , tCreationDate :: C f LocalTime
+  } deriving stock Generic
+    deriving anyclass Beamable
+
+type TargetRec = TargetT Identity
+
+deriving stock instance Show TargetRec
+deriving stock instance Eq TargetRec
+
+instance Table TargetT where
+  data PrimaryKey TargetT f = TargetId (C f (SqlSerial Int32))
+    deriving stock (Generic)
+    deriving anyclass Beamable
+  primaryKey = TargetId . tTargetId
+
+deriving stock instance Show (PrimaryKey TargetT Identity)
+deriving stock instance Eq (PrimaryKey TargetT Identity)
+
+--------------------------
+-- Compound
+--------------------------
+
+data CompoundT f = CompoundRec
+  { cCompoundId :: C f (SqlSerial Int32)
+  , cName :: C f Text
+  , cCreationDate :: C f LocalTime
+  , cChemsoftLink :: C (Nullable f) Text
+  } deriving stock Generic
+    deriving anyclass Beamable
+
+type CompoundRec = CompoundT Identity
+
+deriving stock instance Show CompoundRec
+deriving stock instance Eq CompoundRec
+
+instance Table CompoundT where
+  data PrimaryKey CompoundT f = CompoundId (C f (SqlSerial Int32))
+    deriving stock (Generic)
+    deriving anyclass Beamable
+  primaryKey = CompoundId . cCompoundId
+
+deriving stock instance Show (PrimaryKey CompoundT Identity)
+deriving stock instance Eq (PrimaryKey CompoundT Identity)
+
+--------------------------
+-- Experiment File
+--------------------------
+
+data ExperimentFileT f = ExperimentFileRec
+  { efExperimentFileId :: C f (SqlSerial Int32)
+  , efProjectId :: C f Int32
+  , efMethodologyId :: C (Nullable f) Int32
+  , efUploadDate :: C f LocalTime
+  , efMeta :: C (Nullable f) (PgJSON FileMetadata)
+  , efDescription :: C f Text
+  , efName :: C f Text
+  , efContents :: C f ByteString
+  } deriving stock Generic
+    deriving anyclass Beamable
+
+type ExperimentFileRec = ExperimentFileT Identity
+
+deriving stock instance Show ExperimentFileRec
+deriving stock instance Eq ExperimentFileRec
+
+instance Table ExperimentFileT where
+  data PrimaryKey ExperimentFileT f = ExperimentFileId (C f (SqlSerial Int32))
+    deriving stock (Generic)
+    deriving anyclass Beamable
+  primaryKey = ExperimentFileId . efExperimentFileId
+
+deriving stock instance Show (PrimaryKey ExperimentFileT Identity)
+deriving stock instance Eq (PrimaryKey ExperimentFileT Identity)
+
+--------------------------
+-- Experiment
+--------------------------
+
+data ExperimentT f = ExperimentRec
+  { eExperimentId :: C f (SqlSerial Int32)
+  , eExperimentFileId :: C f Int32
+  , eCompoundId :: C f Int32
```

Comment:
![image](https://user-images.githubusercontent.com/2556409/111868855-68686c00-898d-11eb-8553-eb2c0f5f59a8.png)

Here they suggest using `PrimaryKey` to refer to primary keys of other tables. Shouldn't `eCompoundId` have type `PrimaryKey CompoundT f`? The same question applies to a bunch of other places in this module.

Reply:
I can't remember all the things that made me stop doing this. But as I remember it doesn't give you any advantages and if foreign key consists of two or more params you also can't write `PrimaryKey` for it. So, in the past I decided that it is better to not think about it as you still able to join tables

Reply:
Okay, I didn't use this stuff before, so I don't know whether it's useful or not. Let's come to this topic later if we think that `PrimaryKey` here is better for some reason.